app/controllers/profile_controller.rb

```diff
@@ -1,5 +1,26 @@
 class ProfileController < ApplicationController
 
-  def index
+  def me
+    @user = current_user
+    @user_questions = answered_questions @user
+    render 'show'
   end
+
+  def show
+    @user = User.find_by(username: params[:username])
+
+    if @user
+      @user_questions = answered_questions @user
+    else
+      render 'not_found'
```

Comment:
Essa abordagem tem alguns problemas pois, apesar de ser um 404, retorna como uma view normal (status 200). Também pega uma view que está dentro do escopo de perfis, seria legal ter uma página pública pra redirecionar sempre que houver um erro. Uma forma de fazer isso é essa aqui: https://rubyinrails.com/2018/02/26/rails-render-404-not-found-from-controller-action/